packages/workbox-window/Workbox.mjs

```diff
@@ -0,0 +1,417 @@
+    if (activeSw && urlsMatch(activeSw.scriptURL, this._scriptUrl)) {
+      this._sw = activeSw;
+      this._swPromiseResolver(activeSw);
+    }
+
+    // We set this at registration time rather than instantiation time
+    // in the unlikely event that a page creates multiple Workbox instances
+    // but staggers their registration.
+    this._controllingSw = navigator.serviceWorker.controller;
+
+    // If a SW with the same URL was already controlling the page, let it know
+    // the window is ready to receive messages.
+    if (this._controllingSw &&
+        urlsMatch(this._controllingSw.scriptURL, this._scriptUrl)) {
+      this._controllingSw.postMessage({
+        type: 'WINDOW_READY',
+        meta: 'workbox-window',
+      });
+    }
+
+    if (process.env.NODE_ENV !== 'production') {
+      logger.log('Successfully registered service worker.');
+
+      if (this._controllingSw) {
+        if (urlsMatch(this._controllingSw.scriptURL, this._scriptUrl)) {
+          logger.debug('A service worker with the same script URL is ' +
+              'already controlling this page. This service worker will ' +
+              'remain active unless an update is found.');
+        } else {
+          logger.debug('A service worker with a different script URL is ' +
+              'currently controlling the page.');
+          logger.debug('Fetching the new script...');
+        }
+      }
+
+      // If there's an active and waiting service worker before the
+      // `updatefound` event fires, it means there was a waiting service worker
+      // in the queue before this one was registered.
+      if (reg.waiting && reg.active) {
+        logger.warn('A service worker was already waiting to activate before ' +
+            'this service worker was registered...');
+      }
+
+      const currentPageIsOutOfScope = () => {
+        const scopeUrl = new URL(
+            this._swOptions.scope || this._scriptUrl, document.baseURI);
+        const scopeUrlBasePath = new URL('./', scopeUrl.href).pathname;
+        return !location.pathname.startsWith(scopeUrlBasePath);
+      };
+      if (currentPageIsOutOfScope()) {
+        logger.warn('The current page is not in scope for the registered ' +
+            'service worker. Was this a mistake?');
+      }
+    }
+
+    reg.addEventListener('updatefound', this._onUpdateFound);
+
+    // Add message listeners.
+    if ('BroadcastChannel' in self) {
+      this._broadcastChannel = new BroadcastChannel('workbox');
+      this._broadcastChannel.addEventListener('message', this._onMessage);
+    }
+    navigator.serviceWorker.addEventListener('message', this._onMessage);
+
+    // Keep track of when registration happened, so it can be used in the
+    // `this._onUpdateFound` heuristic. Also use the presence of this
+    // property as a way to see if `.register()` has been called.
+    this._registrationTime = performance.now();
+
+    // Expose and return the registration object.
```

Comment:
Drop "and return", since nothing is being `return`ed.

Reply:
Good catch. I was originally returning the registration, but then opted not to do that.